lib/ui.rb

```diff
@@ -24,7 +25,7 @@ def menu(game_options)
   def get_game_mode(game_options)
     mode = input.gets.chomp
     if (1..game_options.length).include? mode.to_i || mode == "q"
```

Comment:
I would try to bake in the predicate of this if-statement into the GameOptionsMapper.
As it stands, you have two places that care about the `1..game_options.length` and do `mode.to_i`.